qemu.go

```diff
@@ -217,32 +217,46 @@ func (q *qemu) appendFSDevices(devices []ciaoQemu.Device, podConfig PodConfig) [
 	return devices
 }
 
-func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig PodConfig) []ciaoQemu.Device {
+func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig *PodConfig) []ciaoQemu.Device {
 	serial := ciaoQemu.SerialDevice{
 		Driver: ciaoQemu.VirtioSerial,
 		ID:     "serial0",
 	}
 
 	devices = append(devices, serial)
 
+	if podConfig.Console == "" {
+		podConfig.Console = fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+	}
+
+	console := ciaoQemu.CharDevice{
+		Driver:   ciaoQemu.Console,
+		Backend:  ciaoQemu.Socket,
+		DeviceID: "console0",
+		ID:       "charconsole0",
+		Path:     podConfig.Console,
+	}
+
+	devices = append(devices, console)
+
 	for i, c := range podConfig.Containers {
-		var console ciaoQemu.CharDevice
-		if c.Interactive == false || c.Console == "" {
-			consolePath := fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+		// Need to add an offset because of the console created for the pod.
+		idx := i + 1
 
+		if c.Interactive == false || c.Console == "" {
 			console = ciaoQemu.CharDevice{
 				Driver:   ciaoQemu.Console,
 				Backend:  ciaoQemu.Socket,
-				DeviceID: fmt.Sprintf("console%d", i),
-				ID:       fmt.Sprintf("charconsole%d", i),
-				Path:     consolePath,
+				DeviceID: fmt.Sprintf("console%d", idx),
+				ID:       fmt.Sprintf("charconsole%d", idx),
+				Path:     fmt.Sprintf("%s/%s/%s/console.sock", runStoragePath, podConfig.ID, c.ID),
```

Comment:
@sboeuf In that case can we add a debug parameter to our runtime to allow a direct console interface into the guest OS using something along the lines of 
https://github.com/jcvenegas/cc-oci-runtime/commit/3334a2bc7c915e1f631bc85355f1ce20ea984200

This will provide a way to debug any launch or agent failure.